### ISSUE TITLE
ML track design

### ISSUE THREAD

**Issue by theSage21** [PyJaipur/SoA#40]
@kabirnagpal

**Comment by anubhavsharma430** [comment]
What are we gonna do in this track can anyone elaborate ?

**Comment by theSage21** [comment]
@anubhavsharma430 this was discussed in yesterday's volunteer call. Perhaps reading through the MoM would help you?

Otherwise I'll wait for @kabirnagpal to respond here.

https://github.com/PyJaipur/PyJaipur/issues/100#issue-636855460

**Comment by kabirnagpal** [comment]
My team and I will be designing the ML track. Most of the content can be found  [here](https://github.com/kabirnagpal/SoA-ML-14). We'll be working on updating some datasets as the links provided in the notebooks might be old. As the notebooks were created while I was a mentor last year, I feel that it will fit the track with a possibility of collaboration between the two communities.
Also please update me about the format you're expecting from us.  
[100DaysofML](https://github.com/kabirnagpal/100DaysofML)

**Comment by theSage21** [comment]
@kabirnagpal each track is simply a folder with markdown files in it. You can refer to https://github.com/PyJaipur/SoA/tree/master/soa/tracks/python for the naming convention of the markdown files and how to populate the `__init__.py` file for the track.

**Comment by kabirnagpal** [comment]
Okay. My team is working on improving the content. Until when is it required?
Also, the notebooks are already created and changes will be reflected in them with time. 
Changes include adding some EDA, changing some dataset and depreciated methods, adding mathematical expression written from scratch and adding blog links to follow. 
The notebooks can be seen [here](https://github.com/kabirnagpal/SoA-ML-14).

**Comment by anubhavsharma430** [comment]
@kabirnagpal  May i join your team because to have some good ideas which i want to contribute. Even i also want to add up some things in the syllabus as i think its mandatory for interview and task given before interview point of view so that a ML Enthusiast can work in industry. Also I'm started improving my GitHub day by day you can see [here](https://github.com/anubhavsharma430/Classical-Machine-Learning). It done in near 1 months. for others to get help learning ML in pretty much mathematical and Library ways.

**Comment by theSage21** [comment]
@kabirnagpal I think we could aim to have a basic track layout by this Sunday. That way if any changes need to happen we can review it on the volunteer call https://github.com/PyJaipur/PyJaipur/issues/103

If no changes are needed we can then move on and publish the first version of the ML track. Later on as content keeps changing or as feedback comes in we can update the track details. I don't expect the track to change much since the actual content is outside this repo.

@anubhavsharma430 @kabirnagpal I think track development discussions can be held in this issue itself. For the content development, we can discuss things wherever the content resides, be it [https://github.com/kabirnagpal/SoA-ML-14 or https://github.com/anubhavsharma430/Classical-Machine-Learning .

I propose that we can discuss things on this issue if they are about:

- Decide if something should be put in the ML track. For example, should we include 'matrix math' in this track?
- Decide ordering of tasks in a track.
- Decide completion tests for each task.
- How to teach/illustrate a particular task. For example, how to best make a person understand a Core Vector Machine.

**Comment by kabirnagpal** [comment]
Sure that can work, @theSage21. 
Also @anubhavsharma430 please have a look at my [resources](https://github.com/kabirnagpal/SoA-ML-14) and recommend changes in Track. 
As I was informed in the thread that we need to add just the markdown files for the track, we can add hyperlinks to both the repositories, specifically for Scratch code we can refer to @anubhavsharma430 repository, while for the use of Scikit Learn we can refer mine.
Also, certain Higher-order mathematics need not share with the newcomers, as it will scare them off. Rather we can give hyperlinks to certain blogs for them to read.

**Comment by anubhavsharma430** [comment]
Sure I'll see to it @kabirnagpal 
we can introduce maths in a certain way so it gives comfort to the newcomers. What you think ?@theSage21 @kabirnagpal

**Comment by theSage21** [comment]
@anubhavsharma430 it might be easier to make a separate math track. That way concepts can be isolated cleanly. I'm a little hesitant to do that however since by doing that we'd end up becoming a khan academy clone.

**Comment by kabirnagpal** [comment]
What if we take a poll at the very beginning and ask about the domains from which the mentees are? If the majority of them is a non-computer background, we can go with a slower pace and introduce Mathematical interpretation as in the code of @anubhavsharma430. If the majority is from computer background, we can go with the code part as in my repo, followed by some exiting project capstone. What so ever we can always provide them with an opportunity to learn both by introducing hyperlink to other's repo, in our notebook, while discussing a topic.
@theSage21

**Comment by theSage21** [comment]
@kabirnagpal We're aiming to make SoA a continuous platform. When you think of it like that, there's no clear "beginning" since the site would be up all the time.

An alternative would be to monitor users on the site and have a place for them to tell us what tracks are "missing" in their opinion. This can be done in an A/B fashion so that we don't waste anyone's time on too many polls.

This idea touches another filed I wanted to explore. Can tracks change their content based on what all a person already "knows".
For example a basic python track would change if a person is on "windows" or "ubuntu".

**Comment by kabirnagpal** [comment]
So how should we go about that? including both maths and code? @theSage21

**Comment by theSage21** [comment]
@kabirnagpal as I [mentioned earlier](https://github.com/PyJaipur/SoA/issues/40#issuecomment-644100046), as a first step we can introduce the material you already have via an introductory ML track. As our requirements change, we can adapt to them.

**Comment by kabirnagpal** [comment]
Sure.

**Comment by anubhavsharma430** [comment]
Ok Sure
I think the Polling option is good. @theSage21 , @kabirnagpal

**Comment by kabirnagpal** [comment]
@theSage21 I've made a pull request please check if it's fine.
We'll be adding 8th markdown file soon.
@anubhavsharma430 Please look into the repository. I haven't heard from you in a while. Are you still working on contributing to our track?

**Comment by theSage21** [comment]
@kabirnagpal I merged that PR and fixed a few errors it gave rise to. You can test out the track at https://pyj-soa-master.herokuapp.com/task/ml1

If any changes are needed, simply submit another PR and I'll merge it.

**Comment by kabirnagpal** [comment]
For week 1 I'm unable to view the bullet points in the development website but it's visible on Github.
Also, these links are being redirected to ML track which I designed and not the community. 
How'll we redirect students opting for ML track to us?

**Comment by theSage21** [comment]
@kabirnagpal 

1. GitHub fixes a lot of mistakes in markdown. Sadly we don't have the same features. To make that appear as a list I think you'll need to introduce an empty line before starting the list [here](https://github.com/PyJaipur/SoA/blame/master/soa/tracks/ml/1.md#L4).
2. The links were to the content you designed. Can you specify what part should be redirected to the community? I had initially thought that we can add a community link in the track description.
![image](https://user-images.githubusercontent.com/7693265/87750280-a3236e80-c818-11ea-8c8a-3cca9ff03c3f.png)

**Comment by kabirnagpal** [comment]
@theSage21 yeah that would be great. Also along with the Hyperlink to the resource, we can add a hyperlink to the community like this,
In case of doubt or discussion join the community [here](https://t.me/ml_code_for_100_days).

**Comment by theSage21** [comment]
How about adding a button for each task which redirects you to the telegram group?

![image](https://user-images.githubusercontent.com/7693265/87752274-981f0d00-c81d-11ea-8ce1-285cff3cbd59.png)

**Comment by kabirnagpal** [comment]
That looks even better.

**Comment by theSage21** [comment]
Okay then. I'll try to get this done today/tomorrow.
